packages/i18n/src/load-i18n.ts

```diff
@@ -1,49 +1,34 @@
 import type { MessageFormatElement } from '@formatjs/icu-messageformat-parser';
 
-import moment from 'moment';
-import {
-  mergeMessages,
-  mapLocaleToMomentLocale,
-  mapLocaleToIntlLocale,
-} from './utils';
+import { mergeMessages, mapLocaleToIntlLocale } from './utils';
 
-type MomentImportData = {
-  default: moment.Locale;
-};
 export type I18NImportData = {
   default: Record<string, string> | Record<string, MessageFormatElement[]>;
 };
 export type MergedMessages =
   | Record<string, string>
   | Record<string, MessageFormatElement[]>;
 
-const getMomentChunkImport = (locale: string): Promise<MomentImportData> => {
-  const momentLocale = mapLocaleToMomentLocale(locale);
-  switch (momentLocale) {
-    case 'de':
-      return import(
-        /* webpackChunkName: "i18n-moment-locale-de" */ 'moment/locale/de'
-      );
-    case 'es':
-      return import(
-        /* webpackChunkName: "i18n-moment-locale-es" */ 'moment/locale/es'
-      );
-    case 'fr':
-      return import(
-        /* webpackChunkName: "i18n-moment-locale-fr" */ 'moment/locale/fr'
-      );
-    case 'zh-cn':
-      return import(
-        /* webpackChunkName: "i18n-moment-locale-zh-cn" */ 'moment/locale/zh-cn'
-      );
-    case 'ja':
-      return import(
-        /* webpackChunkName: "i18n-moment-locale-ja" */ 'moment/locale/ja'
-      );
-    default:
-      return import(
-        /* webpackChunkName: "i18n-moment-locale-en-gb" */ 'moment/locale/en-gb'
-      );
+const MOMENT_INCLUDED_METADATA_LOCALE = 'en';
+
+const loadMomentLocaleMetadata = async (locale: string) => {
+  // Default English is already included in moment default package
+  // so we don't need to load it explicitly
+  if (locale === MOMENT_INCLUDED_METADATA_LOCALE) return Promise.resolve();
+
+  try {
+    await import(
+      /* webpackChunkName: "i18n-moment-locale-[name]" */ `../compiled-data/moment/locales/${locale}.js`
+    );
+  } catch (error) {
+    // If not-found locale was a regional one, we use the main language by default,
+    // otherwise we use the default locale (already included).
+    const [language, region] = locale.split('-');
+    if (region && language !== MOMENT_INCLUDED_METADATA_LOCALE) {
```

Comment:
If `locale` is an english regional one (eg: `en-CO`) and it does not exist, we don't want to load the main one ((it would be `en`) as that one is already bundled with `moment.js`.